### PR TITLE
challenge: add two integers challenge

### PR DESCRIPTION
**Pull request by aoyeoku**
# Add Two Integers Challenge

## Challenge Description
Given two integers `num1` and `num2`, return the `sum` of the two integers.

## Examples
1. **Input**: `num1 = 12, num2 = 5`  
   **Output**: `17`  
   **Explanation**: num1 is 12, num2 is 5, and their sum is 12 + 5 = 17, so 17 is returned

2. **Input**: `num1 = -10, num2 = 4`  
   **Output**: `-6`  
   **Explanation**: num1 + num2 = -6, so -6 is returned.

## Constraints
- `-100 <= num1, num2 <= 100`

## Helpful Resources
- [Add Two Integers - LeetCode](https://leetcode.com/problems/add-two-integers/description/)

### Files

- [x] The file name describes the function's behavior
- [x] There is a module docstring in the function file
- [x] The test file's name matches the function file name -
  `/tests/test_file_name.py`
- [x] There is a module docstring in the tests file

### Unit Tests

- [x] The test class has a helpful name in PascalCase
- [x] The test class has a docstring
- Every unit test has
  - [x] A helpful name
  - [x] A clear docstring
  - [x] Only one assertion
  - [x] There is no logic in the unit test
- [x] All tests pass
- [x] There are tests for defensive assertions
- [x] There are tests for boundary cases

### Function Docstring

- [x] The function's behavior is described
- The function's arguments are described:
  - [x] Type
  - [x] Purpose
  - [x] Other assumptions (eg. if it's a number, what's the expected range?)
- The return value is described
  - [x] Type
  - [x] Other assumptions are documented
- The defensive assertions are documented using `Raises:`
  - [x] Each assumption about an argument is checked with an assertion
  - [x] Each assertion checks for _only one_ assumption about the argument
- [ ] Include 3 or more (passing!) doctests

### The Function

- [x] The function's name describes it's behavior
- [x] The function's name matches the file name
  - _It's ok to have extra helper functions if necessary, like with mergesort_
- [x] The function has correct type annotations
- [x] The function is not called at the top level of the function file 
  - _Recursive solutions **can** call the function from **inside** the function body_

### Do's

- [x] Variable names help to understand the strategy
- [x] Any comments are clear and describe the strategy
- [x] Lines of code are spaced to help show different stages of the strategy

### Don'ts

- [x] The function's strategy _is not_ described in any docstrings or tests
- [x] Comments explain the _strategy_, **not** the _implementation_
- [x] The function _does not_ have more comments than code
  - If it does, consider finding a new strategy or a simpler implementation

## Implementation

- [x] The code passes the formatting checks
- [x] The code passes all Ruff linting checks
- [x] The code has no (reasonable) Pylint errors
  - In code review, you can decide when fixing a Pylint error is helpful and
    when it's too restricting.
- [x] Variables are named with snake_case
- [x] Variable names are clear and helpful
- [x] The code follows the strategy as simply as possible
- [x] The implementation is as simple as possible given the strategy
- [x] There are no commented lines of code
- [ ] There are no `print` statements anywhere
- [x] The code includes defensive assertions
- [x] Defensive assertions include as little logic as possible

### CODE REVIEW

**Comment by doctorbanu** [comment]
Hi, Abbiihjah 💫 Can you please edit your description? We should customize the name and about parts, and add your function's behavior. Thank you in advance 😄

**Comment by doctorbanu** [comment]
> Hi, Abbiihjah 💫 Can you please edit your description? We should customize the name and about parts, and add your function's behavior. Thank you in advance 😄
Here is a example for you. These are from my factorial challenge:

name: Factorial Calculation
about: A function that calculates the factorial of a non-negative integer using recursion
Behavior
This function calculates the factorial of a non-negative integer n using recursion. The factorial of a number is the product of all positive integers less than or equal to that number.

Input: The function accepts a single integer n, where n must be a non-negative integer (i.e., n >= 0).
Output: The function returns the factorial of n, which is the product of all integers from 1 to n. If n is 0, the factorial is defined as 1.
Error Handling: If the input is not an integer or if n is negative, the function raises an AssertionError.
The factorial is calculated recursively, where the base case is when n = 0, returning 1, and the recursive case multiplies n with the factorial of n - 1.

File: solutions/function_adding_two_integers.py [hunk 1]
```diff
@@ -0,0 +1,38 @@
+#!/usr/bin/env python3
+# -*- coding: utf-8 -*-
+"""
+This module provides a utility function for basic arithmetic operations.
+
+Functions:
+    adding_two_integers(num1: int, num2: int) -> int:
+        Calculates the adding_two_integers of two integers.
+
+Example:
```
Comment (GeehanAli) [inline solutions/function_adding_two_integers.py:10]: The function should include at least three passing doctests. Please add more to meet this requirement.

File: solutions/function_adding_two_integers.py [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+    To use the `adding_two_integers` function:
+
+    >>> result = adding_two_integers(5, 7)
+    >>> print(result)
+    12
+"""
+
```
Comment (GeehanAli) [inline solutions/function_adding_two_integers.py:17]: There are no comments explaining the strategy, but it's not a major issue in this case since the function is straightforward.

**Review by GeehanAli — COMMENTED** [review]
The function and tests meet the checklist requirements.
The code passes the formatting checks, Ruff linting checks and has no Pylint errors. The work is well-organized, clearly written, and demonstrates excellent attention to detail. Please review my comments and make the changes so I can approve the review.
Great work!

File: solutions/function_adding_two_integers.py [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+Functions:
+    adding_two_integers(num1: int, num2: int) -> int:
+        Calculates the adding_two_integers of two integers.
+
+Example:
+    To use the `adding_two_integers` function:
+
+    >>> result = adding_two_integers(5, 7)
+    >>> print(result)
```
Comment (GeehanAli) [inline solutions/function_adding_two_integers.py:14]: There is a print() statement in the module docstring. Using print for output is not permitted at this stage of our study.

File: solutions/function_adding_two_integers.py (hunk 1 above)
Comment (aoyeoku) [reply solutions/function_adding_two_integers.py:10]: Ok, I will add these now.

File: solutions/function_adding_two_integers.py (hunk 3 above)
Comment (aoyeoku) [reply solutions/function_adding_two_integers.py:1226]: I have reviewed and removed the print() statement, please check again, Geehan, thank you.

**Review by GeehanAli — APPROVED** [review]
All requested changes have been made, and everything looks clear and well-executed. Great work!
 I'll proceed to approve the review and close the PR.

**Comment by GeehanAli** [comment]
All requested changes have been made, and everything looks clear and well-executed. Great work! 
I'll proceed to approve the review and close the PR.
